### PR TITLE
adding a proper HTTP response draining function

### PR DESCRIPTION
**Pull request by kris-watts-gravwell**
and checking that it is used

### CODE REVIEW

**Comment by kris-watts-gravwell** [comment]
@david-fritz-gravwell PTAL

File: client/utils.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	ErrMaxBodyDrained = errors.New("too many response bytes in body, closing")
+)
+
+// drainResponse will drain up to 4MB of data then close the response Body.
+// We do this so that http requests can re-use connects as per docs.
+func drainResponse(resp *http.Response) {
+	if resp.Body == nil {
+		return
+	}
+	var nw nilWriter
```
Comment (david-fritz-gravwell) [inline client/utils.go:31]: why not use ioutil.Discard as the writer here?

**Review by david-fritz-gravwell — CHANGES_REQUESTED** [review]

**Review by david-fritz-gravwell — APPROVED** [review]
LGTM

File: client/utils.go (hunk 1 above)
Comment (kris-watts-gravwell) [reply client/utils.go:31]: a limitedWriter + io.Discard would be equivalent.  This felt just as simple.

**Pull request merged**